components/Button.js

```diff
@@ -1,23 +1,55 @@
 import React, { PureComponent } from 'react';
-import { TouchableOpacity } from 'react-native';
+import { LayoutAnimation, TouchableOpacity } from 'react-native';
 import { connectAnimation } from '@shoutem/animation';
 import { connectStyle } from '@shoutem/theme';
+import { Spinner } from './Spinner';
 
 class Button extends PureComponent {
-  render() {
-    // The underlayColor is not a valid RN style
-    // property, so we have to unset it here.
-    const style = {
-      ...this.props.style,
+  componentDidUpdate(prevProps) {
+    const { inProgress: prevInProgress } = prevProps;
+    const { inProgress } = this.props;
+
+    if (prevInProgress !== inProgress) {
+      LayoutAnimation.easeInEaseOut();
+    }
+  }
+
+  resolveStyle() {
+    const { inProgress, style } = this.props;
+
+    if (!inProgress) {
+      return style;
+    }
+
+    return {
+      ...style,
+      ...style?.inProgress,
+      borderRadius: style?.inProgress?.borderRadius || 50,
+      minWidth: style?.inProgress?.minWidth || 30,
+      width: style?.inProgress?.width || 50,
+      paddingLeft: style?.inProgress?.paddingLeft || 0,
+      paddingRight: style?.inProgress?.paddingRight || 0,
+      flex: style?.inProgress?.flex || 1,
+      alignSelf: style?.inProgress?.alignSelf || 'center',
+      opacity: style?.inProgress?.opacity || 0.5,
     };
-    delete style.underlayColor;
+  }
+
+  render() {
+    const { children, inProgress, style, ...otherProps } = this.props;
+
+    const resolvedStyle = this.resolveStyle();
```

Comment:
you would need to memo this if you would go on with different styling, which we won't do